lib/types/operators/Modulo.cpp

```diff
@@ -0,0 +1,211 @@
+#include "../../../includes/types/JsAny.hpp"
+
+JS::Any JS::Any::operator%(const JS::Any& other) const {
+    switch (this->value.index()) {
+        case NUMBER:
+            switch (other.value.index()) {
+                case NUMBER:
+                    return JS::Any(std::fmod(std::get<double>(this->value), std::get<double>(other.value)));
+                case STRING:
+                    return JS::Any(
+                        std::fmod(std::get<double>(this->value), std::stod(std::get<Rope>(other.value).toString())));
+                case BOOL:
+                    return JS::Any(
+                        std::fmod(std::get<double>(this->value), static_cast<double>(std::get<bool>(other.value))));
+                default:
+                    return JS::Any(std::numeric_limits<double>::quiet_NaN());
+            }
+        case STRING:
+            switch (other.value.index()) {
+                case NUMBER:
+                    return JS::Any(
+                        std::fmod(std::stod(std::get<Rope>(this->value).toString()), std::get<double>(other.value)));
+                case STRING:
+                    return JS::Any(std::fmod(std::stod(std::get<Rope>(this->value).toString()),
+                                             std::stod(std::get<Rope>(other.value).toString())));
+                case BOOL:
+                    return JS::Any(std::fmod(std::stod(std::get<Rope>(this->value).toString()),
+                                             static_cast<double>(std::get<bool>(other.value))));
+                default:
+                    return JS::Any(std::numeric_limits<double>::quiet_NaN());
+            }
+        case BOOL:
+            switch (other.value.index()) {
+                case NUMBER:
+                    return JS::Any(
+                        std::fmod(static_cast<double>(std::get<bool>(this->value)), std::get<double>(other.value)));
+                case STRING:
+                    return JS::Any(std::fmod(static_cast<double>(std::get<bool>(this->value)),
+                                             std::stod(std::get<Rope>(other.value).toString())));
+                case BOOL:
+                    return JS::Any(std::fmod(static_cast<double>(std::get<bool>(this->value)),
+                                             static_cast<double>(std::get<bool>(other.value))));
+                default:
+                    return JS::Any(std::numeric_limits<double>::quiet_NaN());
+            }
+        default:
```

Comment:
missing case null because null % null is returning NaN but null % 12 is returning 0